bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafPanelHandler.java

```diff
@@ -187,71 +207,156 @@ private void sendRenderedEffectCommand(Command command) throws NanoleafException
         int blue = rgbPercent[2].toBigDecimal().divide(BigDecimal.valueOf(100), 2, BigDecimal.ROUND_HALF_UP)
                 .multiply(new BigDecimal(255)).intValue();
 
-        Effects effects = new Effects();
-        Write write = new Write();
-        write.setCommand("display");
-        write.setAnimType("static");
-        String panelID = this.thing.getConfiguration().get(NanoleafBindingConstants.CONFIG_PANEL_ID).toString();
-        write.setAnimData(String.format("1 %s 1 %d %d %d 0 10", panelID, red, green, blue));
-        write.setLoop(false);
-        effects.setWrite(write);
         Bridge bridge = getBridge();
         if (bridge != null) {
-            NanoleafControllerConfig config = ((NanoleafControllerHandler) bridge.getHandler()).getControllerConfig();
-            Request setNewRenderedEffectRequest = OpenAPIUtils.requestBuilder(httpClient, config, API_EFFECT,
-                    HttpMethod.PUT);
-            setNewRenderedEffectRequest.content(new StringContentProvider(gson.toJson(effects)), "application/json");
-            OpenAPIUtils.sendOpenAPIRequest(setNewRenderedEffectRequest);
+            Effects effects = new Effects();
+            Write write = new Write();
+            write.setCommand("display");
+            write.setAnimType("static");
+            String panelID = this.thing.getConfiguration().get(NanoleafBindingConstants.CONFIG_PANEL_ID).toString();
+            @Nullable BridgeHandler handler = bridge.getHandler();
+            if (handler!=null) {
+                NanoleafControllerConfig config = ((NanoleafControllerHandler) handler).getControllerConfig();
+                // Light Panels and Canvas use different stream commands
+                if (config.deviceType.equals(NanoleafBindingConstants.CONFIG_DEVICE_TYPE_LIGHTPANELS)) {

```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)